core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -49,13 +49,13 @@ class TransactionsTest extends KafkaServerTestHarness {
   val nonTransactionalConsumers = Buffer[KafkaConsumer[Array[Byte], Array[Byte]]]()
 
   override def generateConfigs: Seq[KafkaConfig] = {
-    TestUtils.createBrokerConfigs(numServers, zkConnect, true).map(KafkaConfig.fromProps(_, serverProps()))
+    TestUtils.createBrokerConfigs(numServers, zkConnect).map(KafkaConfig.fromProps(_, serverProps()))
   }
 
   @Before
   override def setUp(): Unit = {
     super.setUp()
-    val numPartitions = 3
+    val numPartitions = 1
```

Comment:
Having a single partition makes the test assertions simpler. I'll revert this and only use one partition on the new test case.